### PR TITLE
fix: transfer bugfix

### PR DESCRIPTION
**Pull request by nuanyang233**
## Description

Closes #MASK-718 #MASK-748

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

**Comment by wenluomask** [comment]
Checked.

File: packages/web3-shared/evm/hooks/useContract.ts [hunk 1]
```diff
@@ -19,19 +19,21 @@ export function createContract<T extends BaseContract>(web3: Web3, address: stri
  * EthereumService in the background page and decode the result of calls automatically
  * @param address
  * @param ABI
+ * @param chainId
  */
 export function useContract<T extends BaseContract>(address = '', ABI: AbiItem[] = [], chainId?: ChainId) {
-    const web3 = useWeb3({ chainId })
+    const web3 = useWeb3(chainId ? { chainId } : {})
```
Comment (hanszhoou) [inline packages/web3-shared/evm/hooks/useContract.ts:25]: Is there difference between `const web3 = useWeb3({ chainId })` and `const web3 = useWeb3(chainId ? { chainId } : {})` here?

File: packages/mask/src/plugins/Trader/SNSAdaptor/trader/Trader.tsx [hunk 2]
```diff
@@ -114,12 +111,28 @@ export function Trader(props: TraderProps) {
...
+                ? createERC20Token(
+                      chainId,
+                      defaultInputCoin.contract_address,
+                      defaultInputCoin.decimals,
+                      defaultInputCoin.name,
+                      defaultInputCoin.symbol,
+                  )
+                : undefined,
+        })
+    }, [defaultInputCoin])
```
Comment (hanszhoou) [inline packages/mask/src/plugins/Trader/SNSAdaptor/trader/Trader.tsx]: does it depend on `chainId`?

File: packages/mask/src/extension/popups/pages/Wallet/Transfer/Transfer1559.tsx [hunk 3]
```diff
@@ -419,6 +419,8 @@ export const Transfer1559 = memo<Transfer1559Props>(({ selectedAsset, openAssetM
         [selectedAsset, transferCallback, registeredAddress, Utils],
     )
 
+    console.log(error)
```
Comment (hanszhoou) [inline packages/mask/src/extension/popups/pages/Wallet/Transfer/Transfer1559.tsx]: remove this line

File: packages/mask/src/plugins/Trader/SNSAdaptor/trader/Trader.tsx [hunk 4]
```diff
@@ -48,12 +48,13 @@ const useStyles = makeStyles()(() => {
...
     coin?: Coin
+    defaultInputCoin?: Coin
     tokenDetailed?: FungibleTokenDetailed
     chainId?: ChainId
 }
 
 export function Trader(props: TraderProps) {
-    const { coin, tokenDetailed, chainId: targetChainId } = props
+    const { coin, tokenDetailed, chainId: targetChainId, defaultInputCoin } = props
     const { decimals } = tokenDetailed ?? coin ?? {}
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/SNSAdaptor/trader/Trader.tsx:57]: Does it need to be removed?

File: packages/mask/src/plugins/Trader/SNSAdaptor/trader/Trader.tsx (hunk 2 above)
Comment (guanbinrui) [reply packages/mask/src/plugins/Trader/SNSAdaptor/trader/Trader.tsx]: I think so, chainId was used.

File: packages/web3-shared/evm/hooks/useContract.ts (hunk 1 above)
Comment (nuanyang233) [reply packages/web3-shared/evm/hooks/useContract.ts:25]: Yes, a chainId of undefined will cause a different result

File: packages/mask/src/plugins/Trader/SNSAdaptor/trader/Trader.tsx (hunk 2 above)
Comment (nuanyang233) [reply packages/mask/src/plugins/Trader/SNSAdaptor/trader/Trader.tsx]: Just for business. Adding the chainId will cause a bug in the dependency

**Comment by wenluomask** [comment]
Also fixed https://mask.atlassian.net/browse/MASK-643 here.

**Comment by wenluomask** [comment]
Checked.

**Review by Jack-Works — APPROVED** [review]

**Comment by AinuX** [comment]
Checked

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**